pinc/RandomRule.inc

```diff
@@ -41,9 +41,8 @@ class RandomRule
             WHERE id = %d
         ", $id);
 
-        $result = mysqli_query(DPDatabase::get_connection(), $sql);
-        if(!$result)
-        {
+        $result = DPDatabase::query($sql, FALSE);
```

Comment:
why FALSE on the throwing error?

Reply:
Because I wanted to preserve the type of exception thrown in case it was significant. This one throws a UnexpectedValueException, whereas DPDatabase::query throws a DBQueryError.

Reply:
If the exception type isn't significant (and it looks like it isn't because there are only 2 (unrelated) places in the codebase that `catch(UnexpectedValueException $e)`), I agree we could leave it up to ::query to throw the exception.

Reply:
The `UnexpectedValueException` is wrong too (and I wrote it -- blame me). The intent was probably to indicate that a value someone passed in was wrong -- but if that were actually the case the SQL would return zero rows so `$result` would still be valid. I would remove the `$result` test and let `DBDatabase::query()` raise.

Reply:
Done!